trailblazer/store/utils/executors.py

```diff
@@ -0,0 +1,115 @@
+"""API for executors."""
+
+import logging
+from typing import List
+
+from trailblazer.constants import TOWER_STATUS, TrailblazerStatus
+from trailblazer.store.utils.tower import (
+    TowerProcess,
+    TowerTask,
+    TowerTaskResponse,
+    TowerWorkflowResponse,
+)
+from trailblazer.store.utils.tower_client import TowerApiClient
+
+LOG = logging.getLogger(__name__)
+
+
+class ExecutorAPI:
+    """Handles communication between executors and trailblazer."""
+
+    pass
+
+
+class TowerAPI(ExecutorAPI):
+    """Handles communication with tower."""
+
+    def __init__(self, executor_id: str, dry_run: bool = False):
+        self.executor_id: str = executor_id
```

Comment:
`workflow_id` would work BUT `workflow` is confusing as I would think of it as the object containing all workflow information. I was thinking that if at some point we do a SlurmAPI class, it would make sense to keep the same attribute name and then executor_id might be more appropriate since this is the id of the executor (tower or slurm) performing the analysis.

Reply:
I see `executor_id` is a bit vague to me. Are any of `case_id`, `job_id` or `analysis_id` applicable?

Reply:
refactored to workflow_id